Intersect.Client/Entities/Entity.cs

```diff
@@ -1411,29 +1411,21 @@ protected virtual bool ShouldDrawHpBar
         {
             get
             {
-                if (!ShouldNotDrawHpBar || IsHovered)
+                if (ShouldNotDrawHpBar)
                 {
-                    return true;
+                    return false;
                 }
 
-                return GetType() == typeof(Entity) && Globals.Database.NpcOverheadHpBar;
+                return (GetType() == typeof(Entity) && Globals.Database.NpcOverheadHpBar) ||
```

Comment:
Check IsHovered first, then shield, then HP, and only then should you check the entity type.

This should also be broken up into multiple if statements so it isn't such a long line.